### ISSUE TITLE
[Backend] IFOs could return continuation

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#599]
In our current design the `apply` method of each IFO does not return anything (void).

As far as I can see, we could use the return value to return the next IFO to be evaluated (or null) if the result is already available. The return type would then be e.g.  `Cont$Bool`  or  `Cont$Obj`.

This has some possible benefits: 
- We avoid having to write to context.continuation (which is currently a megamorphic, if you will.)
- We avoid the cast that is currently necessary when reading from context.continuation.

In other words, we replace the continuation pointer inside  `Context` with a return value. Since this return value can be appropriately typed (e.g.  `Cont$Bool`  or  `Cont$Obj`) we also avoid a cast.

**Comment by magnus-madsen** [comment]
We are replacing IFOs by trampolines.
